components/fxa-client/src/device.rs

```diff
@@ -12,20 +12,37 @@ use crate::{
         CommandData, DeviceUpdateRequest, DeviceUpdateRequestBuilder, PendingCommand,
         UpdateDeviceResponse,
     },
-    FirefoxAccount, IncomingDeviceCommand,
+    util, CachedResponse, FirefoxAccount, IncomingDeviceCommand,
 };
 use serde_derive::*;
 use std::collections::{HashMap, HashSet};
 
+// An devices response is considered fresh for `DEVICES_FRESHNESS_THRESHOLD` ms.
+const DEVICES_FRESHNESS_THRESHOLD: u64 = 60_000; // 1 minute
+
 impl FirefoxAccount {
     /// Fetches the list of devices from the current account including
     /// the current one.
-    pub fn get_devices(&self) -> Result<Vec<Device>> {
+    pub fn get_devices(&mut self) -> Result<Vec<Device>> {
```

Comment:
in profile we also have `ignore_cache`, would we need something like this here as well https://github.com/mozilla/application-services/blob/f3fe12419e64b2c77f40398c8e54ca4a30672873/components/fxa-client/src/profile.rs#L21

Reply:
Desktop does have a similar `ignore_cache` facility FWIW

Reply:
@rfk From reading the comments in the bug I wasn't sure if this should be omitted until explicitly called for. I've added it.